test/metabase/lib/query_test.cljc

```diff
@@ -506,3 +508,26 @@
                                     "1969-10-12"
                                     "1971-10-12"]]}]}
               (lib/query mp query))))))
+
+#?(:clj
+   (deftest ^:synchronized cache-test
+     (let [query      (lib/query meta/metadata-provider (meta/table-metadata :orders))
+           viz-cols   lib.metadata.calculation/visible-columns-method
+           calls      (atom 0)
+           exp-fields (into #{} cat
+                            [(map #(meta/id :orders %)   (meta/fields :orders))
+                             (map #(meta/id :people %)   (meta/fields :people))
+                             (map #(meta/id :products %) (meta/fields :products))])]
+       (is (=? {} (-> query meta :lib/__cache)))
```

Comment:
This assertion is a bit confusing, what does it want to express? (I expected a stronger assertion here.)

Reply:
That the cache is attached where it's expected to be, and is a maplike. That it's working is proved below.

I've added more context with `testing` clauses.

Reply:
I also added some more permutations that eg. on modifying the query the cache is separate from the original.